<a href="https://colab.research.google.com/github/chrislee973/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/LS_DS_123_Confidence_Intervals_Assignment_No_Answers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [ ]:
#Import statements

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
df = pd.read_csv( 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv')

In [ ]:
#Print the top 5 rows
df.head()

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [ ]:
#Calculations here

mean = df['Caffeine'].mean()
std = df['Caffeine'].std(ddof=1)
n=len(df)
st_err = std/np.sqrt(n)

print(f"The mean of the sample is {mean}, the standard deviation is {std}, and the standard error is {st_err}.")


The mean of the sample is 37.9402, the standard deviation is 5.243756828216712, and the standard error is 0.7415792024250598.


Sentence or two summary.

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [ ]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df=221)
print('t_star =', t_star)

t_star = 1.9707562704880786


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [ ]:
#Calculations here

t_star = t.ppf(.975, df= n-1 )

moe = t_star* st_err

moe

1.49025919960566

The margin of error is 1.49 oz of caffeine for a 95% confidence interval.

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [ ]:
#Calculations here

confidence_interval = (mean-moe, mean+moe)

confidence_interval

(36.44994080039434, 39.43045919960566)

###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [ ]:
#Calculations here

t.interval(alpha=.95, df=n-1, loc=mean, scale = st_err)

(36.44994080039434, 39.43045919960566)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

Yes, they do and should match.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

We are 95% confident that the population mean of caffeine content lies within the interval between 36.44 oz of caffeine and 39.43 oz of caffeine. 

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [ ]:
#Calculations here

t_star = t.ppf(1-(1-.9)/2, df = n-1)
moe = t_star*st_err

confidence_interval_90 = (mean-moe, mean+moe)

confidence_interval_90

(36.696904726749196, 39.1834952732508)

We are 90% certain that the population mean of caffeine content lies within the range of 36.69 oz and 39.18 oz of caffeine. This is more precise than the 95% confidence interval. 

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [ ]:
#Calculations here

t_star = t.ppf(1-(1-.99)/2, df = n-1)
moe = t_star*st_err

confidence_interval_99 = (mean-moe, mean+moe)

confidence_interval_99

(35.95280335285685, 39.92759664714315)

This estimate is more accurate, but less precise than the 95% confidence interval.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Answer here.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

The margin of error would shrink because the margin of error is calculated by multiplying t-value with standard error. And when the sample size increases the standard error shrinks, so all else being equal the margin of error would also shrink.

In [ ]:
#Calculations here.

Compare MEs here.

